### PR TITLE
Deprecates support for el7/python2.7 and adds support for python 3.11

### PR DESCRIPTION
**Pull request by lorengordon**

### CODE REVIEW

**Comment by lorengordon** [comment]
go codebuild go

**Comment by lorengordon** [comment]
go codebuild go

File: .travis.yml [hunk 1]
```diff
@@ -14,12 +14,12 @@ env:
     - RELEASE_VERSION=$(grep -E '^current_version' $TRAVIS_BUILD_DIR/.bumpversion.cfg | sed 's/^.*= //')
     - PRIOR_VERSION=$(git describe --abbrev=0 --tags)
 python:
-  - 2.7
   - 3.6
   - 3.7
   - 3.8
   - 3.9
   - 3.10
+  - 3.11
```
Comment (ferricoxide) [inline .travis.yml:22]: Not ready to work on 3.12, yet?

File: .travis.yml [hunk 2]
```diff
@@ -72,19 +72,19 @@ jobs:
   include:
     - stage: test
       name: lint
-      python: 3.10
+      python: 3.11
```
Comment (ferricoxide) [inline .travis.yml:75]: Not ready to work on 3.12, yet?

File: .travis.yml [hunk 3]
```diff
@@ -72,19 +72,19 @@ jobs:
...
       env:
         - TOXENV=check
 
     - stage: test
       name: docs
-      python: 3.10
+      python: 3.11
```
Comment (ferricoxide) [inline .travis.yml:81]: Not ready to work on 3.12, yet?

File: .travis.yml [hunk 4]
```diff
@@ -72,19 +72,19 @@ jobs:
...
       env:
         - TOXENV=docs
 
     - stage: test
       name: build standalone
-      python: 3.10
+      python: 3.11
```
Comment (ferricoxide) [inline .travis.yml:87]: Not ready to work on 3.12, yet?

File: .travis.yml [hunk 5]
```diff
@@ -142,7 +142,7 @@ jobs:
     - stage: deploy
       name: deploy to pypi
       if: type != pull_request AND (branch = main OR tag is present)
-      python: 3.10
+      python: 3.11
```
Comment (ferricoxide) [inline .travis.yml:145]: Not ready to work on 3.12, yet?

File: setup.cfg [hunk 6]
```diff
@@ -21,32 +21,29 @@ classifiers =
...
     Operating System :: Microsoft :: Windows
     Programming Language :: Python
-    Programming Language :: Python :: 2.7
     Programming Language :: Python :: 3
     Programming Language :: Python :: 3.6
     Programming Language :: Python :: 3.7
     Programming Language :: Python :: 3.8
     Programming Language :: Python :: 3.9
     Programming Language :: Python :: 3.10
+    Programming Language :: Python :: 3.11
```
Comment (ferricoxide) [inline setup.cfg:30]: Not ready to work on 3.12, yet?

**Review by ferricoxide — APPROVED** [review]
I know it's early for 

> Not ready to work on 3.12, yet?

Just wanted to raise it "just in case"

**Comment by lorengordon** [comment]
> I know it's early for
> 
> > Not ready to work on 3.12, yet?
> 
> Just wanted to raise it "just in case"

Can always add it later, presuming it is available for travis-ci and azure pipelines already, as well as the apt/yum repos.

**Pull request merged**